File: MantenimientoDaxDirectorio/MantenimientoDirectorio/Module1.vb
Option Strict Off
Option Explicit On
Imports DattCom
Module Module1
    Public Orden As String = ""

    Public Sub Mainn()
        Try
            'If datosEmpresa.strConxAdcom = "" Then varbleComun.cargar.iniciar("", "")
            If Len(Trim(datosEmpresa.strConxAdcom)) = 0 Then Exit Sub
        Catch ee As Exception
            MsgBox("excepción mainn: " & ee.Message)
        End Try
        Emp.CargarValores()
    End Sub

    Public Function ValidaId(ByRef NumeroId As String, ByRef TipoId As String, ByVal Persona As String) As Boolean
        Dim Largo As Short
        On Error Resume Next
        ValidaId = False
        If TipoId = "P" Then
            ValidaId = True
            Exit Function
        ElseIf TipoId = "F" Then
            If NumeroId = "9999999999999" Then ValidaId = True
            Exit Function
        End If
        Largo = Len(NumeroId)
        If Not IsNumeric(NumeroId) Then Exit Function
        If TipoId = "R" Then
            If Largo <> 13 Or Mid(NumeroId, 11) <> "001" Then Exit Function
        ElseIf TipoId = "C" Then
            If Largo <> 10 Then Exit Function
        End If
        Largo = Val(Mid(NumeroId, 3, 1))
        ValidaId = True
        If TipoId = "C" Then
            '           ValidaId = Validador.validarCedula(NumeroId)
        ElseIf Largo >= 0 And Largo <= 5 Then
            '            ValidaId = Validador.validarRUCNaturales(NumeroId)
        ElseIf Largo = 6 Then
            '            ValidaId = Validador.validarRUCPublicas(NumeroId)
        ElseIf Largo = 9 Then
            '            ValidaId = Validador.validarRUCPrivadas(NumeroId)
        End If
        'UPGRADE_NOTE: El objeto Validador no se puede destruir hasta que no se realice la recolección de los elementos no utilizados. Haga clic aquí para obtener más información: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="6E35BFF6-CD74-4B09-9689-3E1A43DF8969"'
        '        Validador = Nothing
    End Function

    Public Function ClienteMovimiento(ByRef codigo As String) As Boolean
        Dim cod As String
        Dim ConxAdcom As New SqlClient.SqlConnection(datosEmpresa.strConxAdcom)
        Dim rs As SqlClient.SqlDataReader
        ClienteMovimiento = False
        Try
            ConxAdcom.Open()
            If ConxAdcom.State = 1 Then
                cod = " SELECT Doc_codper " & " From AdcDoc " & " Where Doc_codper = '" & codigo & "' "
                Dim comm As New SqlClient.SqlCommand(cod, ConxAdcom)
                rs = comm.ExecuteReader
                If rs.Read Then ClienteMovimiento = True
                rs.Close()
                comm.Dispose()
                cod = " SELECT idempleado " & " From rolliq " & " Where idempleado = '" & codigo & "' "
                comm = New SqlClient.SqlCommand(cod, ConxAdcom)
                rs = comm.ExecuteReader
                If rs.Read Then ClienteMovimiento = True
                rs.Close()
                comm.Dispose()
            End If
            ConxAdcom.Close()
        Catch ee As Exception
            MsgBox("excepción Mandir_Clmov : " & ee.Message)
        End Try
    End Function
End Module

File: MantenimientoDaxDirectorio/MantenimientoDirectorio/Module1.cs
using System;
using DattCom;
using Microsoft.VisualBasic;

namespace directMnt
{
	static class Module1
	{
		public static string Orden = "";

		public static void Mainn()
		{
			try
			{
				// If datosEmpresa.strConxAdcom = "" Then varbleComun.cargar.iniciar("", "")
				if (Len(Strings.Trim(datosEmpresa.strConxAdcom)) == 0)
					return;
			}
			catch (Exception ee)
			{
				Interaction.MsgBox("excepción mainn: " + ee.Message);
			}
			Emp.CargarValores();
		}

		public static bool ValidaId(ref string NumeroId, ref string TipoId, string Persona)
		{
			bool ValidaIdRet = default;
			short Largo;
			;
#error Cannot convert OnErrorResumeNextStatementSyntax - see comment for details
			/* Cannot convert OnErrorResumeNextStatementSyntax, CONVERSION ERROR: Conversion for OnErrorResumeNextStatement not implemented, please report this issue in 'On Error Resume Next' at character 673


						Input:
								On Error Resume Next

						 */
			ValidaIdRet = false;
			if (TipoId == "P")
			{
				ValidaIdRet = true;
				return ValidaIdRet;
			}
			else if (TipoId == "F")
			{
				if (NumeroId == "9999999999999")
					ValidaIdRet = true;
				return ValidaIdRet;
			}
			Largo = (short)Strings.Len(NumeroId);
			if (!Information.IsNumeric(NumeroId))
				return ValidaIdRet;
			if (TipoId == "R")
			{
				if (Largo != 13 | Strings.Mid(NumeroId, 11) != "001")
					return ValidaIdRet;
			}
			else if (TipoId == "C")
			{
				if (Largo != 10)
					return ValidaIdRet;
			}
			Largo = (short)Math.Round(Conversion.Val(Strings.Mid(NumeroId, 3, 1)));
			ValidaIdRet = true;
			if (TipoId == "C")
			{
			}
			// ValidaId = Validador.validarCedula(NumeroId)
			else if (Largo >= 0 & Largo <= 5)
			{
			}
			// ValidaId = Validador.validarRUCNaturales(NumeroId)
			else if (Largo == 6)
			{
			}
			// ValidaId = Validador.validarRUCPublicas(NumeroId)
			else if (Largo == 9)
			{
				// ValidaId = Validador.validarRUCPrivadas(NumeroId)
			}

			return ValidaIdRet;
			// UPGRADE_NOTE: El objeto Validador no se puede destruir hasta que no se realice la recolección de los elementos no utilizados. Haga clic aquí para obtener más información: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="6E35BFF6-CD74-4B09-9689-3E1A43DF8969"'
			// Validador = Nothing
		}

		public static bool ClienteMovimiento(ref string codigo)
		{
			bool ClienteMovimientoRet = default;
			string cod;
			var ConxAdcom = new System.Data.SqlClient.SqlConnection(datosEmpresa.strConxAdcom);
			System.Data.SqlClient.SqlDataReader rs;
			ClienteMovimientoRet = false;
			try
			{
				ConxAdcom.Open();
				if ((int)ConxAdcom.State == 1)
				{
					cod = " SELECT Doc_codper " + " From AdcDoc " + " Where Doc_codper = '" + codigo + "' ";
					var comm = new System.Data.SqlClient.SqlCommand(cod, ConxAdcom);
					rs = comm.ExecuteReader();
					if (rs.Read())
						ClienteMovimientoRet = true;
					rs.Close();
					comm.Dispose();
					cod = " SELECT idempleado " + " From rolliq " + " Where idempleado = '" + codigo + "' ";
					comm = new System.Data.SqlClient.SqlCommand(cod, ConxAdcom);
					rs = comm.ExecuteReader();
					if (rs.Read())
						ClienteMovimientoRet = true;
					rs.Close();
					comm.Dispose();
				}
				ConxAdcom.Close();
			}
			catch (Exception ee)
			{
				Interaction.MsgBox("excepción Mandir_Clmov : " + ee.Message);
			}

			return ClienteMovimientoRet;
		}
	}
}